addons/nuxeo-ai-core/elements/admin-ai-export.js

```diff
@@ -38,142 +62,137 @@ Polymer({
+  _visibleChanged(newVal) {
+    if (newVal) {
+      this.$.aiExportStatus.execute();
+    } else if (!newVal && clearTimeout !== undefined) {
+      clearTimeout(this._timeOut);
     }
-    if (val.type === 'terms') {
-      let toReturn = '';
-      for (let i = 0, len = val.value.length; i < len; i++) {
-        toReturn += ` ${val.value[i].key} (${val.value[i].docCount})`;
-      }
-      return toReturn;
-    }
-    return JSON.stringify(val.value);
   },
 
-  _filter(items) {
-    return items && items.filter((item) => !(item.type === 'total' || item.type === 'count'));
+  _statusReceived(resp) {
+    this.statuses = resp.detail.response.value;
+    // defines if the tab is currently active if so, we want to poll more often
+    if (this.visible) {
+      this._timeOut = setTimeout(this._poll, 2000, this.$.aiExportStatus);
+    }
   },
 
-  _computeLabel(theStats, fieldName) {
-    const foundItem = theStats && theStats.find((item) => item.type === fieldName);
-    return foundItem && this.i18n(`admin.ai.export.${fieldName}`, foundItem.numericValue);
+  _poll(func) {
+    return func.execute();
   },
 
-  _recompute() {
-    this.$.aiStats.params = {
-      query: this.query,
-      inputs: this.inProps,
-      outputs: this.outProps,
-    };
-    this.$.aiStats
-      .execute()
-      .then((response) => {
-        if (response === undefined || response.length === 0) {
-          this.fire('notify', { message: this.i18n('admin.ai.stats.none') });
-        }
-      })
-      .catch(() => {
-        this.fire('notify', { message: this.i18n('admin.ai.stats.error') });
-      });
+  _updateItem(e) {
+    this.$.currentDoc = e.detail.value;
   },
 
   _export() {
+    if (!this._canExport()) {
+      this.fire('notify', { message: 'Type and Model are required parameters' });
+      return;
+    }
+    const props = this.$.currentDoc.properties;
+    const inputs = props['ai_model:inputs'];
+    const input = inputs.map((e) => e.name).join(',');
+
+    const outputs = props['ai_model:outputs'];
+    const output = outputs.map((e) => e.name).join(',');
+
+    const query =
+      `SELECT * FROM Document WHERE ecm:isProxy = 0` +
+      ` AND ecm:isVersion = 0 AND ecm:isTrashed = 0 AND ecm:primaryType = '${this.typeName}'`;
     this.$.aiExport.params = {
-      query: this.query,
-      inputs: this.inProps,
-      outputs: this.outProps,
-      split: this.splitProp,
+      query,
+      inputs: input,
+      outputs: output,
     };
+
     this.$.aiExport
       .execute()
       .then((response) => {
-        this._exportText = this.i18n('admin.ai.export.id', response.value);
         this.fire('notify', { message: this.i18n('admin.ai.export.success') });
+        return response.value;
       })
       .catch(() => {
-        this._exportText = '';
         this.fire('notify', { message: this.i18n('admin.ai.export.error') });
       });
   },
+
+  _canExport() {
+    return this.typeName.length !== 0 && this.$.currentDoc !== undefined;
```

Comment:
Can we export if `this.$.currentDoc` is `null`? If not:
```suggestion
    return this.typeName.length !== 0 && this.$.currentDoc != undefined;
```

Reply:
we cannot. I don't see how it can be null. undefined - yes

Reply:
Since this is exposed as a public property, someone from outside can simply set it to `null`. Consider doing the check for `!= undefined` instead or setting it as a [read-only property](https://polymer-library.polymer-project.org/2.0/docs/devguide/properties#read-only).

Reply:
`!=` eslint won't allow it. added explicit check

Reply:
It's the equivalent to != null, which is allowed.